### PR TITLE
New rule: shopify_infra_abuse.yml

### PR DESCRIPTION
**Pull request by aidenmitchell**
# Description

Attackers have been observed using myshopify.com links to link to malicious pages. Also observed in Dropbox impersonation campaigns.

# Associated samples

- https://platform.sublimesecurity.com/hunts/43eb8902-9c93-4901-8e2d-9f4b609af03b

### CODE REVIEW

**Comment by aidenmitchell** [comment]
No hits

File: detection-rules/shopify_infra_abuse.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: "Shopify infrastructure abuse"
+description: "Attackers have been observed using myshopify.com links to bypass domain reputation checks."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          .href_url.domain.root_domain == "myshopify.com"
+  
+          // negate legitimate Shopify newsletters
```
Comment (zoomequipd) [inline detection-rules/shopify_infra_abuse.yml:10]: should this section be within the `any(body.links, ...` or on it's own outside of that loop?

File: detection-rules/shopify_infra_abuse.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          and (
+            not any(headers.domains, .root_domain == "klaviyomail.com")
+            and not any(body.links,
+                        .href_url.domain.root_domain in ("klclick2.com")
+            )
+            and not any(headers.domains,
+                        .root_domain not in ("shopify.com", "myshopify.com")
```
Comment (zoomequipd) [inline detection-rules/shopify_infra_abuse.yml:17]: nested `not` here.  was this intentional? 

maybe each of these 3 conditions just needs to be it's own `and not ...`?

File: detection-rules/shopify_infra_abuse.yml (hunk 1 above)
Comment (aidenmitchell) [reply detection-rules/shopify_infra_abuse.yml]: Re-reading this months later, I'm not sure what was going on here. I've restructured it to make a lot more sense.

**Review by zoomequipd — APPROVED** [review]

**Pull request merged**